### PR TITLE
feat(template): support global vars as param initializer e.g. templat…

### PR DESCRIPTION
**Pull request by gpolychronis**
…e((state = globalState) => {...})

*Global variable is handled like complex default values i.e. a function or a constructor
*Added an example, accessible at http://localhost:5000/examples/variableArgument/

### CODE REVIEW

File: src/trax/compiler/parser.ts [hunk 1]
```diff
@@ -194,7 +194,8 @@ export function parse(src: string, filePath: string, options?: ParserOptions): (
                 }, skipProperty = false;
 
                 m.forEachChild((c) => {
-                    if (c.kind === SK.Identifier) {
+                    if (c.kind === SK.Identifier && !prop.name) {
+                        //First SK.Identifier is property name. Second one might be a variable as default value.
```
Comment (b-laporte) [inline src/trax/compiler/parser.ts]: trax changes should be submitted first on the trax repo (I will share it with you)

File: src/xjs/parser.ts [hunk 2]
```diff
@@ -320,7 +320,6 @@ export async function parse(tpl: string, filePath = "", lineOffset = 0, columnOf
             } else if (lookup(V_RW)) {
                 advance(V_RW);
                 nd.defaultValue = currentText();
-                console.log("TODO: support default value as variable: ", nd.defaultValue);
```
Comment (b-laporte) [inline src/xjs/parser.ts]: ... and xjs changes should be submitted on the xjs repo :-) (I will also share it with you)

**Review by b-laporte — CHANGES_REQUESTED** [review]
Hi - thanks for the PR
As most of the changes touch trax, the code changes should be submitted to the trax repo first.
Regarding the example, I was thinking of updating the pages example instead of creating a new one (each page could be given a global state that they could update through a button for instance)
Thx

**Comment by gpolychronis** [comment]
@b-laporte 
Do you need this PR? If not may I decline it?

**Pull request closed without merging**